validator/client/src/test/java/tech/pegasys/teku/validator/client/restapi/apis/PostVoluntaryExitTest.java

```diff
@@ -14,27 +14,73 @@
 package tech.pegasys.teku.validator.client.restapi.apis;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_BAD_REQUEST;
 import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_FORBIDDEN;
 import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_INTERNAL_SERVER_ERROR;
 import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_NOT_FOUND;
 import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_OK;
 import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_UNAUTHORIZED;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.EPOCH;
+import static tech.pegasys.teku.infrastructure.http.RestApiConstants.PUBKEY;
 import static tech.pegasys.teku.infrastructure.restapi.MetadataTestUtil.getResponseStringFromMetadata;
 import static tech.pegasys.teku.infrastructure.restapi.MetadataTestUtil.verifyMetadataErrorResponse;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
+import java.util.Optional;
 import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.restapi.StubRestApiRequest;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 import tech.pegasys.teku.spec.Spec;
 import tech.pegasys.teku.spec.SpecMilestone;
 import tech.pegasys.teku.spec.TestSpecFactory;
 import tech.pegasys.teku.spec.datastructures.operations.SignedVoluntaryExit;
+import tech.pegasys.teku.spec.datastructures.operations.VoluntaryExit;
 import tech.pegasys.teku.spec.util.DataStructureUtil;
+import tech.pegasys.teku.validator.client.VoluntaryExitDataProvider;
 
 public class PostVoluntaryExitTest {
-  private final PostVoluntaryExit handler = new PostVoluntaryExit();
+  private final VoluntaryExitDataProvider provider = mock(VoluntaryExitDataProvider.class);
+  private final PostVoluntaryExit handler = new PostVoluntaryExit(provider);
   private final Spec spec = TestSpecFactory.createMinimal(SpecMilestone.CAPELLA);
   private final DataStructureUtil dataStructureUtil = new DataStructureUtil(spec);
+  private final StubRestApiRequest request = new StubRestApiRequest(handler.getMetadata());
+
+  final VoluntaryExit message = new VoluntaryExit(UInt64.valueOf(123), UInt64.ZERO);
+  final SignedVoluntaryExit signedVoluntaryExit =
+      new SignedVoluntaryExit(message, dataStructureUtil.randomSignature());
+
+  @Test
+  void noEpochQueryParamProvided_shouldGiveValidSignedVoluntaryExit()
```

Comment:
i think really these are just testing that we should call through to the provider with the parameter values.

The 'validSignedVoluntaryExit' is coming just from the fact that we've generated one and passed one out... so all we're really testing here is that the params are correctly passed to the provider...

`shouldCallProviderWhenEpochNotProvided`
`shouldCallProviderWhenEpochProvided`

These might be better names...